types/connect/index.d.ts

```diff
@@ -1,94 +1,163 @@
+         * The route is always terminated at a path separator (`/`) or a dot (`.`) character. This means the
+         * given routes `/foo/` and `/foo` are the same and both will match requests with the URLs
+         * `/foo`, `/foo/`, `/foo/bar`, and `/foo.bar`, but not match a request with the URL `/foobar`.
+         *
+         * The route is matched in a case-insensitive manner.
+         * @param fn A plain function, a Node.js HTTP server instance or another Connect app instance.
+         */
+        use(route: string, fn: (
+            /** The Node.js http request object. */ req: Request,
+            /** The Node.js http response object. */ res: http.ServerResponse,
+            /** A function to call to invoke the next middleware. */ next: NextFunction
+        ) => void): Server;
+
+        /**
+         * Register an "error-handling" middleware. These are middleware where the
+         * function takes exactly 4 arguments. When a middleware passes an error to `next`, the app will
+         * proceed to look for the error middleware that was declared after that middleware and invoke it,
+         * skipping any error middleware above that middleware and any non-error middleware below.
+         * @param fn A plain function, a Node.js HTTP server instance or another Connect app instance.
+         */
+        use(fn: (
+            /** The error that occurred. */ err: any,
+            /** The Node.js http request object. */ req: Request,
+            /** The Node.js http response object. */ res: http.ServerResponse,
+            /** A function to call to invoke the next middleware. */ next: NextFunction
+        ) => void): Server;
+
+        /**
+         * Mount an "error-handling" middleware on the specified route.
+         * @param route An optional path string that is matched against the beginning of the incoming request URL.
+         * This allows for basic routing.
+         *
+         * The route is always terminated at a path separator (`/`) or a dot (`.`) character. This means the
+         * given routes `/foo/` and `/foo` are the same and both will match requests with the URLs
+         * `/foo`, `/foo/`, `/foo/bar`, and `/foo.bar`, but not match a request with the URL `/foobar`.
+         *
+         * The route is matched in a case-insensitive manner.
+         * @param fn A plain function, a Node.js HTTP server instance or another Connect app instance.
+         */
+        use(route: string, fn: (
+            /** The error that occurred. */ err: any,
+            /** The Node.js http request object. */ req: Request,
+            /** The Node.js http response object. */ res: http.ServerResponse,
+            /** A function to call to invoke the next middleware. */ next: NextFunction
+        ) => void): Server;
+
+        /** Register a middleware. */
         use(fn: HandleFunction): Server;
+        /** Mount a middleware on the specified route. */
         use(route: string, fn: HandleFunction): Server;
```

Comment:
I originally intended to replace them with the overloads above, with the direct explicit parameters of `fn` callbacks visible, because that's more practical to handle in the editor (VSCode) after autocomplete, etc. Also, that allows more accurate documentation on each specific overload.
However, I had to keep the overloads with `HandleFunction` to avoid breaking compatibility in the tests of other packages, which referenced `HandleFunction` interface directly by name. Maybe I should have added `@deprecated` here as well.